model/aspnetcore/metrics.yaml

```diff
@@ -114,3 +125,166 @@ groups:
     attributes:
       - ref: aspnetcore.rate_limiting.result
         requirement_level: required
+
+  # blazor components
+  - id: metric.aspnetcore.components.navigation
+    type: metric
+    metric_name: aspnetcore.components.navigation
+    stability: development
+    brief: Total number of route changes.
+    instrument: counter
+    unit: "{route}"
+    note: |
+      Meter name: `Microsoft.AspNetCore.Components`; Added in: ASP.NET Core 10.0
+    extends: aspnetcore.components.metrics.attributes
+    attributes:
+      - ref: aspnetcore.components.type
+        requirement_level: required
+      - ref: aspnetcore.components.route
+        requirement_level: required
+
+  - id: metric.aspnetcore.components.event.duration
+    type: metric
+    metric_name: aspnetcore.components.event.duration
+    stability: development
+    brief: Duration of processing browser event.
+    instrument: histogram
+    unit: "s"
+    note: |
+      Meter name: `Microsoft.AspNetCore.Components`; Added in: ASP.NET Core 10.0
+    extends: aspnetcore.components.metrics.attributes
+    attributes:
+      - ref: aspnetcore.components.type
+        requirement_level: required
+      - ref: aspnetcore.components.method
+        requirement_level: required
+      - ref: aspnetcore.components.attribute.name
+        requirement_level: required
+
+  - id: metric.aspnetcore.components.event.exceptions
+    type: metric
+    metric_name: aspnetcore.components.event.exceptions
+    stability: development
+    brief: Total number of exceptions during browser event processing.
+    instrument: counter
+    unit: "{exception}"
+    note: |
+      Meter name: `Microsoft.AspNetCore.Components`; Added in: ASP.NET Core 10.0
+    extends: aspnetcore.components.metrics.attributes
+    attributes:
+      - ref: aspnetcore.components.type
+        requirement_level: required
+      - ref: aspnetcore.components.attribute.name
+        requirement_level: required
+      - ref: error.type
+        brief: The full name of exception type.
+        examples: ['System.OperationCanceledException', 'Contoso.MyException']
+        requirement_level: required
+
+  - id: metric.aspnetcore.components.update_parameters.duration
+    type: metric
+    metric_name: aspnetcore.components.update_parameters.duration
+    stability: development
+    brief: Duration of processing component parameters.
+    instrument: histogram
+    unit: "s"
+    note: |
+      Meter name: `Microsoft.AspNetCore.Components.Lifecycle`; Added in: ASP.NET Core 10.0
+    extends: aspnetcore.components.lifecycle.metrics.attributes
+    attributes:
+      - ref: aspnetcore.components.type
+        requirement_level: required
+
+  - id: metric.aspnetcore.components.update_parameters.exceptions
```

Comment:
would be covered by `aspnetcore.components.update_parameters.duration` with error.type